system/Files/FileCollection.php

```diff
@@ -340,6 +340,44 @@ public function retainPattern(string $pattern, ?string $scope = null)
         return $this->removeFiles(array_diff($files, self::matchFiles($files, $pattern)));
     }
 
+    /**
+     * Keeps only the files from the list that match multiple patterns
+     * (within the optional scope).
+     *
+     * @param array<string>       $patterns Array of regex or pseudo-regex strings
```

Comment:
```suggestion
     * @param list<string>       $patterns Array of regex or pseudo-regex strings
```
and run `composer cs-fix` to fix coding style.